.travis.yml

```diff
@@ -10,3 +10,14 @@ python:
   - "nightly" # currently points to 3.6-dev
 # command to run tests
 script: nosetests
+notifications:
+  email:
+    on_success: never
+    on_failure: always
+deploy:
```

Comment:
Current implementation adds this step to all jobs in the matrix. The suggestion is to create an additional stage with a separate job for deployment. And then use if-clause to run that stage only for tagged commits.